File: new_py_pkg/new_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64


class NumberPublisherNode(Node): 

    def __init__(self):
        super().__init__("number_publisher") 
        
        self.declare_parameter("number_to_publish", 2)
        self.declare_parameter("publish_frequency", 1)

        self.publish_frequency = 1 / self.get_parameter("publish_frequency").value

        self.publisher_ = self.create_publisher(Int64, "number",10)
        self.timer_ = self.create_timer(self.publish_frequency, self.publish_number)
        self.get_logger().info("Number Publisher has started!")


    def publish_number(self):
        self.number = self.get_parameter("number_to_publish").value

        msg = Int64()
        msg.data = self.number
        self.publisher_.publish(msg)

    



def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: components_cpp/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

namespace my_namespace {

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher(const rclcpp::NodeOptions &options) : Node("number_publisher", options)
    {
        number_ = 2;

        number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        number_timer_ = this->create_wall_timer(1000ms,
                                                std::bind(&NumberPublisher::publishNumber, this));
        RCLCPP_INFO(this->get_logger(), "Number publisher has been started.");
    }

private:
    void publishNumber()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        number_publisher_->publish(msg);
    }

    int number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::TimerBase::SharedPtr number_timer_;
};
} // namespace my_namespace

#include "rclcpp_components/register_node_macro.hpp"
RCLCPP_COMPONENTS_REGISTER_NODE(my_namespace::NumberPublisher)
```


